### ISSUE TITLE
Implement forgetting about rooms

### ISSUE THREAD

**Issue by babolivier** [matrix-org/dendrite#669]
Full specs: https://matrix.org/docs/spec/client_server/r0.4.0.html#post-matrix-client-r0-rooms-roomid-forget

**Comment by Kegsay** [comment]
Sytests:
```
    × Forgotten room messages cannot be paginated
    × Forgetting room does not show up in v2 /sync
    × Can forget room you've been kicked from
    × Can re-join room if re-invited
```

**Comment by S7evinK** [comment]
This can be closed, implemented in #1572
